components/releases/ReleaseSection.tsx

```diff
@@ -0,0 +1,208 @@
+    }[];
+  };
+}
+
+interface Repository {
+  name: string;
+  releases: {
+    nodes: Release[];
+  };
+}
+
+interface Organization {
+  repositories: {
+    nodes: Repository[];
+  };
+}
+
+interface GitHubResponse {
+  data: any;
+  organization: Organization;
+}
+
+interface Repository {
+  name: string;
+  releases: {
+    nodes: Release[];
+  };
+}
+
+interface Organization {
+  repositories: {
+    nodes: Repository[];
+  };
+}
+
+interface GitHubResponse {
+  organization: Organization;
+}
+
+const ReleaseSection: React.FC = async () => {
+  const accessToken = process.env.GITHUB_PAT;
+
+  if (!accessToken) {
+    if (process.env.NODE_ENV === "development") {
+      console.error("'GITHUB_PAT' is not configured in the environment.");
+      return [];
+    }
+
+    throw "'GITHUB_PAT' is not configured in the environment.";
+  }
+
+  const response = await fetch("https://api.github.com/graphql", {
+    method: "POST",
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      query: `{
+        organization(login: "${process.env.NEXT_PUBLIC_GITHUB_ORG}") {
+          repositories(first: 100) {
+            nodes {
+              name
+              releases(first: 10, orderBy: {field: CREATED_AT, direction: DESC}) {
+                nodes {
+                  name
+                  createdAt
+                  description
+                  url
+                  author {
+                    login
+                    avatarUrl
+                  }
+                  mentions (first: 10) {
+                    nodes {
+                      login 
+                      avatarUrl
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }`,
+    }),
+  });
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const json = (await response.json()) as GitHubResponse;
+  const data = json.data;
+
+  const repositories: Repository[] = data.organization.repositories.nodes;
+  const allReleases: Release[] = [];
+  for (const repository of repositories) {
+    for (const release of repository.releases.nodes) {
+      release.repository = repository.name;
+      allReleases.push(release);
+    }
+  }
+
+  const sortedReleases = allReleases.sort(
+    (a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+  );
+  const latestReleases = sortedReleases.slice(0, 10);
+
+  if (latestReleases.length === 0) {
+    return (
+      <>
+        <div className="w-full h-10 bg-gray-200 dark:bg-gray-700 animate-pulse rounded" />
+        <div className="w-full h-10 bg-gray-200 dark:bg-gray-700 animate-pulse rounded" />
+        <div className="w-full h-10 bg-gray-200 dark:bg-gray-700 animate-pulse rounded" />
+        <div className="w-full h-10 bg-gray-200 dark:bg-gray-700 animate-pulse rounded" />
+      </>
+    );
+  }
```

Comment:
1. If an org. does not have a single release, this would always end up being shown and the user maybe confused that it's loading indefinitely.

2. Use suspense fallback to show loading state instead in the parent component. 

Refer: https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#example